### PR TITLE
metapb: refine store state

### PR DESCRIPTION
**Pull request by rleungx**
This PR introduces `NodeState` to replace `StoreState`, which is confusing sometimes, in the future.

### CODE REVIEW

File: pkg/brpb/brpb.pb.go [hunk 1]
```diff
@@ -61,7 +61,7 @@ func (x CompressionType) String() string {
 	return proto.EnumName(CompressionType_name, int32(x))
 }
 func (CompressionType) EnumDescriptor() ([]byte, []int) {
-	return fileDescriptor_brpb_a90a70ff0dc32c8e, []int{0}
+	return fileDescriptor_brpb_7c3d2f8143177a80, []int{0}
```
Comment (disksing) [inline pkg/brpb/brpb.pb.go:64]: would you like to reset the unrelated files?

Comment (rleungx) [reply pkg/brpb/brpb.pb.go:64]: These differences are generated by running `make go`.

Comment (disksing) [reply pkg/brpb/brpb.pb.go:64]: I know. You can checkout unrelated files to make the diff smaller.

Comment (rleungx) [reply pkg/brpb/brpb.pb.go:64]: got it.

**Review by disksing — APPROVED** [review]

File: proto/metapb.proto [hunk 2]
```diff
@@ -52,6 +63,8 @@ message Store {
     int64 last_heartbeat = 11;
     // If the store is physically destroyed, which means it can never up again.
     bool physically_destroyed = 12;
+    // NodeState is used to replace StoreState which will be deprecated in the future.
```
Comment (BusyJay) [inline proto/metapb.proto:66]: I think you can mark it as deprecated. See https://developers.google.com/protocol-buffers/docs/proto3#options.

File: proto/metapb.proto [hunk 3]
```diff
@@ -26,6 +26,17 @@ enum StoreState {
     Tombstone = 2;
 }
 
+// NodeState is going to replace StoreState to make the state concept more clear.
+// "Up" is devided into "Preparing" and "Serving" stages so that we can better describe the online process.
+// "Removing" is just like previous `Offline` which is more accurate.
+// "Removed" has the same meaning with `Tombstone`.
+enum NodeState {
+    Preparing = 0;
```
Comment (BusyJay) [inline proto/metapb.proto:34]: How about adding docs to explaining each state?

**Review by BusyJay — COMMENTED** [review]
Rest LGTM.

File: proto/metapb.proto (hunk 2 above)
Comment (rleungx) [reply proto/metapb.proto:66]: Yes, once the related feature is finished, I will mark the old state as deprecated.
